data/scrape/scrape.py

```diff
@@ -116,23 +116,38 @@ def _build_scraping_config(
 def main(args: argparse.Namespace) -> None:
     with open(LANGUAGES_PATH, "r", encoding="utf-8") as source:
         languages = json.load(source)
-    codes = list(languages.keys())
-    # Verifies language code for --restriction is valid.
+    codes = frozenset(languages.keys())
     if args.restriction:
         # Cleans entry.
-        keys = re.split(r"[;,\s]+\s*", args.restriction[0].strip(";, "))
-        if not keys[0]:
+        restriction_set = frozenset(
+            re.split(r"[;,\s]+\s*", args.restriction.strip(";, "))
+        )
+        if len(restriction_set) == 1 and not list(restriction_set)[0]:
             # Checks for empty entry.
             logging.fatal("Restriction flag raised but no language provided")
             exit(1)
-        rset = frozenset(args.restriction)
-        lset = frozenset(codes)
-        eset = rset - lset
-        if eset:
-            for key in eset:
+        if not restriction_set.issubset(codes):
+            for key in restriction_set - codes:
                 logging.fatal("%r is not a valid ISO code", key)
             exit(1)
-        codes = list(rset)
+    else:
+        restriction_set = codes
+    if args.exclude:
```

Comment:
I realize that this `if args.exclude` block looks very similar to the `if args.restriction` block above, but I think it's more readable as-is, rather than trying to factor things out that would add more cognitive burden to our future selves.

Reply:
True. I'm not sure what I'd factor out exactly.